sky/resources.py

```diff
@@ -944,9 +941,20 @@ def less_demanding_than(self,
             if other.disk_tier is None:
                 return False
             if self.disk_tier != other.disk_tier:
-                types = ['low', 'medium', 'high']
-                return types.index(self.disk_tier) < types.index(
-                    other.disk_tier)
+                if self.cloud is not None:
+                    self_tier = self.cloud.translate_disk_tier(self.disk_tier)
+                else:
+                    self_tier = self.disk_tier
+                if other.cloud is not None:
+                    other_tier = other.cloud.translate_disk_tier(
+                        other.disk_tier)
+                else:
+                    other_tier = other.disk_tier
+                # best disk tier without knowledge of cloud cannot be compared
+                if resources_utils.DiskTier.BEST in [self_tier, other_tier]:
+                    return False
```

Comment:
This looks very complicated and unnecessary. How about we just let the `best` disk_tier run on any cluster?

Reply:
Did you mean default disk tier to best..?

Reply:
I mean if `self.disk_tier == BEST`, set the `less_demanding_than` to be True?

Reply:
Yeah, we could... Just wondering if it would be a little strange if `sky launch --disk-tier best` works for a cluster launched with `--disk-tier low`, though arguably IIUC most users are using high disk tier. Shall we print some UX warning messages if we fail to compare?

P.S. Just took a look for all usage of `less_demanding_than` and it turns out that this function is always called with `other` as launched resources. Another solution might be to `assert other.cloud is not None` and it will make the comparison way more easier

If none of them sounds necessary to you, I'm okay with just letting this check pass since it might be interpreted as 'use the best tier possible' and for a launched cluster its tier is the best you can get 🤔